### ISSUE TITLE
Navigation arrows are missing on Docsy Example page, and others.  FontAwesome not loading?

### ISSUE THREAD

**Issue by prdsoftware** [google/docsy#1324]
After upgrading to the latest build of Docsy, the little fontawesome arrows next to the table of contents (TOC) are missing.  They used to animate to point right, or down, depending on whether you had the topic open or not.   Now they are just place-holders where the font cannot be loaded.  

Here's my site with the broken font (Navbar missing icons)
https://docs.helpmasterpro.com/docs/workflow/overview/

I notice the Docsy example site they're missing completely also.  See https://www.docsy.dev/docs/

Has something changed, or I need to install something?

Thanks in advance.
Rod

**Comment by LisaFC** [comment]
So the navigation arrows only appear if the foldable sidebar is enabled, which it isn't on the Docsy example site. However, those boxes on your site look like a problem....

**Comment by LisaFC** [comment]
Ok I managed to get your arrows back by switching the font family to Fontawesome 5 in Chrome developer tools, will keep digging. Docsy is currently using Fontawesome 6, which *should* still have free arrow icons...

**Comment by prdsoftware** [comment]
Thanks Lisa.  Yes, that's what I was tracking down also.  I downloaded the latest Docsy example site to see if the arrows worked on it on a fresh install, but I noticed that it too was deploying FontAwesome 5.  I wasn't sure if I needed to refresh the packages or not, or whether I'd just missed something.  The "themes > docsy > assets > vendor > Font-Awesome" folder contains FontAwesome 5 by default.  Is this correct?  Should it be 6?

**Comment by LisaFC** [comment]
So I just cloned the example site but built using Hugo modules (just ran "hugo") and it worked fine - both had arrows and showed that it was using Fontawesome 6 when I checked in the console. @chalin @samiahmedsiddiqui any idea what's going on?

Are you using Git submodules or Hugo modules for your site?

**Comment by prdsoftware** [comment]
Hi @LisaFC and @samiahmedsiddiqui ,

Thanks for your help on this.  I've made the repo public.
https://github.com/prdsoftware/HelpMaster-Documentation

In an attempt to get this working, I removed the assets/vendor folder from source control, but I'm not sure that was the right thing to do?

The netlify build command which I copied from the Docsy Netlify doco gives the following error:

8:22:07 AM: ────────────────────────────────────────────────────────────────
8:22:07 AM:   "build.command" failed                                        
8:22:07 AM: ────────────────────────────────────────────────────────────────
8:22:07 AM: ​
8:22:07 AM:   Error message
8:22:07 AM:   Command failed with exit code 255: cd themes/docsy && git submodule update -f --init && cd ../.. && hugo (https://ntl.fyi/exit-code-255)
8:22:07 AM: ​
8:22:07 AM:   Error location
8:22:07 AM:   In Build command from Netlify app:
8:22:07 AM: Creating deploy upload records
8:22:07 AM:   cd themes/docsy && git submodule update -f --init && cd ../.. && hugo

**Comment by LisaFC** [comment]
Thanks for sharing, I'll take a look!

**Comment by LisaFC** [comment]
OK, so I think you having an assets/vendor folder in the first place might have caused some of the problems, you possibly inadvertently kept it when you copied in the updated theme files  - the Docsy theme hasn't had an assets/vendor directory since version 0.4.0, when we switched to using NPM modules for Fontawesome and Bootstrap.

See release notes here: https://github.com/google/docsy/blob/main/CHANGELOG.md#040

Also if you're using Docsy directly you don't need any of the submodule update stuff in your build command as it doesn't have to fetch any files from the Docsy repository (though I believe you need to run npm install - @chalin? See: https://www.docsy.dev/docs/get-started/other-options/#for-an-existing-site)

Is there a reason why you copied Docsy into your project repo rather than use a submodule or Hugo module? I know some users need to do it, but it can make things a little harder to manage.

**Comment by prdsoftware** [comment]
Thanks @LisaFC for the assistance.

Is the Netlify build command here still valid?  https://www.docsy.dev/docs/deployment/#deployment-with-netlify

This is what I've been following, and it uses a git submodule command.  Is this incorrect now?  I'm pretty confused with all of the instructions on how to set all of this up as there seems to be a lot of different ways, and somewhat conflicting information (at least to me)

Regarding copying Docsy into the repo.... when I was first trialing Docsy, I got it running locally, and then checked everything into Github and I guess Docsy came with it.  Everything worked with Netlify, so I just assumed this was all good.

Sounds like I should remove Docsy from my repository, and use a submodule, or a Hugo module.

Thanks again.

**Comment by LisaFC** [comment]
No problem, and yes, we do have a few setup options! We support them all because
- Some users need to actually have the Docsy theme files in their repo
- Some users prefer to use Docsy as an npm module as that's how they like to manage stuff
- Some users or their contributors can't install Go locally so can't use Hugo modules - so we've maintained support for using Docsy as a Git submodule even though it's a little more complicated.
- Everyone else can use Docsy as a Hugo module (the easiest version if you can install Go).

The Netlify build command in your link is specifically for if you're using Docsy as a Git submodule, I'm going to assign myself an issue to update the docs to clarify this (the Netlify build command for using Hugo modules is just `hugo`, which is nice and simple...) - thanks for bringing it up!

If you can install Go, I would recommend using Docsy as a Hugo module, it's very easy to keep up to date once it's set up. I can take a look in your repo and see what you need to update.

**Comment by chalin** [comment]
Hi all, so is this issue resolved, @LisaFC et all?

**Comment by prdsoftware** [comment]
Thanks @LisaFC and @chalin.  I will be having a go at converting to a Hugo module today and seeing if that helps.

**Comment by prdsoftware** [comment]
OK, after a lot of trial and error and mucking about, I've managed to convert my site to a hugo module and it works well.  

I think a lot of this may have been caused by my misunderstanding the architecture of this, and not being familiar with git, npm, hugo, netlify, go and pretty much the entire stack of tools and languages that make Docsy what it is.  :-)  

For the netlify command, i got it going with the following command:

npm install && hugo

This was the only option that worked for me.

Thanks again @LisaFC  for your assistance.  

Greetings from Canberra, Aus.
Rod
